### ISSUE TITLE
rockstor-bootstrap not starting

### ISSUE THREAD

**Issue by mapero** [rockstor/rockstor-core#1421]
Hi,

the rockstor-bootstrap service stucks at activating. With the help of Steven we were able to narrow the cause. The gunicorn workers timeout:

```
2016-08-06 23:10:08 [3580] [INFO] Starting gunicorn 0.16.1
2016-08-06 23:10:08 [3580] [INFO] Listening at: http://127.0.0.1:8000 (3580)
2016-08-06 23:10:08 [3580] [INFO] Using worker: sync
2016-08-06 23:10:08 [3592] [INFO] Booting worker with pid: 3592
2016-08-06 23:10:08 [3591] [INFO] Booting worker with pid: 3591
2016-08-06 23:12:11 [3580] [CRITICAL] WORKER TIMEOUT (pid:3592)
2016-08-06 23:12:11 [3580] [CRITICAL] WORKER TIMEOUT (pid:3592)
2016-08-06 23:12:11 [5483] [INFO] Booting worker with pid: 5483
2016-08-06 23:14:19 [3580] [CRITICAL] WORKER TIMEOUT (pid:3591)
2016-08-06 23:14:19 [3580] [CRITICAL] WORKER TIMEOUT (pid:3591)
2016-08-06 23:14:19 [7816] [INFO] Booting worker with pid: 7816
2016-08-06 23:14:29 [3580] [CRITICAL] WORKER TIMEOUT (pid:5483)
2016-08-06 23:14:29 [3580] [CRITICAL] WORKER TIMEOUT (pid:5483)
2016-08-06 23:14:29 [8156] [INFO] Booting worker with pid: 8156
2016-08-06 23:16:32 [3580] [CRITICAL] WORKER TIMEOUT (pid:7816)
2016-08-06 23:16:32 [3580] [CRITICAL] WORKER TIMEOUT (pid:7816)
2016-08-06 23:16:32 [10198] [INFO] Booting worker with pid: 10198
2016-08-06 23:18:38 [3580] [CRITICAL] WORKER TIMEOUT (pid:10198)
2016-08-06 23:18:38 [3580] [CRITICAL] WORKER TIMEOUT (pid:10198)
[...]
```

so the bootstrap is not getting the right response:

```
BTRFS device scan complete
Error connecting to Rockstor. Is it running?
Exception occured while bootstrapping. This could be because rockstor.service is still starting up. will wait 2 seconds and try again. Exception: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /api/commands/bootstrap (Caused by <class 'httplib.BadStatusLine'>: '')
Error connecting to Rockstor. Is it running?
Exception occured while bootstrapping. This could be because rockstor.service is still starting up. will wait 2 seconds and try again. Exception: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /api/commands/bootstrap (Caused by <class 'httplib.BadStatusLine'>: '')
Error connecting to Rockstor. Is it running?
Exception occured while bootstrapping. This could be because rockstor.service is still starting up. will wait 2 seconds and try again. Exception: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /api/commands/bootstrap (Caused by <class 'httplib.BadStatusLine'>: '')
[...]
```

The Web UI however is working fine.

More information [in the Forum](https://forum.rockstor.com/t/rockstor-bootstrap-taking-forever/1889)

If you need further information or ideas how to fix it, please let me know.

Thanks.

**Comment by mapero** [comment]
Docker is not removing the btrfs subvolumes when removing an image which results in ophans in the btrfs/subvolumes folder.

I counted over 1000 subvolumes in this directory which I guess was to much for  the bootstrap script. After cleaning the directory manually rockstor boots fine.

Related docker issue:
https://github.com/docker/docker/issues/9939
Even if it is closed, the problem still exists in 1.9.1

**Comment by FroggyFlox** [comment]
Thanks for sharing the fix here... and sorry for the lack of activity on this issue. I'm going to close it now as this would need to be confirmed to still exist in current Rockstor version first.
